avalara_salestax/avalara_api.py

```diff
@@ -185,7 +185,8 @@ def get_tax(self, company_code, doc_date, doc_type, partner_code, doc_code, orig
             line1.Discounted = False
             line1.No = '%d' %line
             line1.ItemCode = received_lines[line].get('itemcode', None)
-            line1.Description = received_lines[line].get('description', None)
+            line1.Description = (received_lines[line].get('description', '')
+                                 ).encode('ascii', 'ignore')
```

Comment:
What is the offending encoding of the passed text?

Reply:
If I understand your question. Encoding that is provided by the Magento line is utf-8, Avalara accepts only ascii. We have built up quite some number of orders stuck because of the ascii error.

Reply:
See, that's the thing. It's not even Avalara that does not accept this input, it's urllib2. When I debug this a little bit (it's easy to reproduce locally by adding an 'é' to the invoice line name in the one test in this module, and then running the tests with the demo settings from the staging database), I see that the message body is actually perfectly valid UTF-8, which is the XML standard encoding and also advertized in the message headers. I think we can solve this like it's 2019 and fix this module so that it works with non ascii characters. Do you want to have another look?

Reply:
Lets have a look when you are here